### PR TITLE
Implement server-side rendering

### PR DESCRIPTION
**Pull request by wchsinyi**
Hi Matt,

I set up server-side rendering using a template.js and react dom stringify

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -1,33 +1,60 @@
...
+import Header from './Header.jsx';
+import Menu from './Menu.jsx';
+import Sidebar from './Sidebar.jsx';
+import exampleData from './exampleData.js';
 
 class App extends React.Component {
-    constructor(props){
+    constructor(props) {
         super(props);
         this.state = ({
```
Comment (matt-violet) [inline client/src/components/App.jsx]: You might have extra parenthesis around your state?

File: client/src/components/Description.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import styled from 'styled-components'
+
+class Description extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = (
```
Comment (matt-violet) [inline client/src/components/Description.jsx]: Extra parenthesis again?

File: client/src/components/Description.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        this.show = this.show.bind(this);
+    }
+    show(){
+        this.setState({
+            show: !this.state.show
+        })
+    }
+    render() {
+        if (this.props.description===undefined){
+            return <div>
```
Comment (matt-violet) [inline client/src/components/Description.jsx]: Perhaps these divs can be aligned either vertically or on the same line?

File: client/src/components/Description.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        }
+        var description = this.props.description.split(' ')
+        var toShow = description.slice(0, 50);
+        if (this.state.show) {
+            return (
+                <div>
+                    {description.join(' ')}
+                    <div onClick={this.show}>
+                        -show less
+                </div>
```
Comment (matt-violet) [inline client/src/components/Description.jsx]: These divs could be better aligned

File: client/src/components/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import Description from './Description.jsx';
+import { review_icon, price_icon, cuisine_icon, crossStreet_icon, diningStyle_icon } from './Icons.jsx';
+import styled from 'styled-components'
```
Comment (matt-violet) [inline client/src/components/Header.jsx]: missing semicolon?

File: client/src/components/Header.jsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+
+const Button = styled.button`
```
Comment (matt-violet) [inline client/src/components/Header.jsx]: is this supposed to have a semicolon at the end?

File: client/src/components/Header.jsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+      item = res.topTags.map( (i, idx)=>(
+          <Button key={idx.toString()}>
+          {i}
+          </Button>
+          
+      ))
+  }
+  return (
+    <Wrapper>
+            <Title>
```
Comment (matt-violet) [inline client/src/components/Header.jsx]: Spacing seems too big between the wrapper and title tags.

File: client/src/components/Header.jsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                star symbols {res.star} review symbol {res.reviewCount} {price_icon} {pricelevel_verbal} {diningStyle_icon} {res.cuisine}
+            </div>
+            <div>
+                Top Tags: {item}
+            </div>
+            <div>
+                <Description description={res.description}/>
+            </div>
+    </Wrapper>
+
```
Comment (matt-violet) [inline client/src/components/Header.jsx]: Unnecessary empty space?

File: client/src/components/Icons.jsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+            </rect>
+            <path d="M14.55,2 C14.2462434,2 14,2.24624339 14,2.55 L14,21 C14,21.5522847 14.4477153,22 15,22 L16,22 C16.5522847,22 17,21.5522847 17,21 L17,17 L18,17 C19.1045695,17 20,16.1045695 20,15 L20,7.45 C20,4.44004811 17.5599519,2 14.55,2 L14.55,2 Z M18,15 L16,15 L16,4.32 C17.2176535,4.88673047 17.9973291,6.10692278 18,7.45 L18,15 Z" id="Shape" fill="#333333" fillRule="nonzero">
+            </path>
+            <path d="M11,2 L6,2 C4.8954305,2 4,2.8954305 4,4 L4,11 C4,12.1045695 4.8954305,13 6,13 L7,13 L7,21 C7,21.5522847 7.44771525,22 8,22 L9,22 C9.55228475,22 10,21.5522847 10,21 L10,13 L11,13 C12.1045695,13 13,12.1045695 13,11 L13,4 C13,2.8954305 12.1045695,2 11,2 Z M11,11 L6,11 L6,4 L7,4 L7,8.5 C7,8.77614237 7.22385763,9 7.5,9 C7.77614237,9 8,8.77614237 8,8.5 L8,4 L9,4 L9,8.5 C9,8.77614237 9.22385763,9 9.5,9 C9.77614237,9 10,8.77614237 10,8.5 L10,4 L11,4 L11,11 Z" id="Shape" fill="#333333" fillRule="nonzero">
+            </path>
+          </g>
+        </g>
+      </g>
+    </svg>
+  </div>);
```
Comment (matt-violet) [inline client/src/components/Icons.jsx]: Your closing parenthesis are on new lines below. Here it is on the same line with the div. Best to be consistent one way or the other?

File: server/index.js [hunk 10]
```diff
@@ -3,41 +3,64 @@ var bodyParser = require('body-parser');
...
+    console.log('listening on port 3001!');
+});
 
 
-// app.get('/api/users', function(req, res) {
-//     var data = JSON.parse(Object.keys(req.query)[0]);
-//     console.log(data.name);
-//     db.getAllUsers(data.name, (err, data) => {
+
+// app.get('/res/:name', function(req, res) {
```
Comment (matt-violet) [inline server/index.js]: The Code Review Checklist says not to include lots of commented out code, so it's just something to consider as you make more progress.

File: client/src/components/Header.jsx (hunk 5 above)
Comment (wchsinyi) [reply client/src/components/Header.jsx]: great catch!

File: client/src/components/App.jsx (hunk 1 above)
Comment (wchsinyi) [reply client/src/components/App.jsx]: great catch!

File: client/src/components/Description.jsx (hunk 2 above)
Comment (wchsinyi) [reply client/src/components/Description.jsx]: Got it!

File: client/src/components/Description.jsx (hunk 3 above)
Comment (wchsinyi) [reply client/src/components/Description.jsx]: resolved :)

File: client/src/components/Description.jsx (hunk 4 above)
Comment (wchsinyi) [reply client/src/components/Description.jsx]: fixed this :)

File: client/src/components/Header.jsx (hunk 7 above)
Comment (wchsinyi) [reply client/src/components/Header.jsx]: Fixed :)

File: server/index.js (hunk 10 above)
Comment (wchsinyi) [reply server/index.js]: fixed :)

File: client/src/components/Icons.jsx (hunk 9 above)
Comment (wchsinyi) [reply client/src/components/Icons.jsx]: fixed :)

**Pull request merged**